.github/workflows/amplify_analytics_pinpoint.yaml

```diff
@@ -78,7 +78,10 @@ defaults:
   run:
     shell: bash
 
-permissions: read-all
+# These permissions are needed to interact with GitHub's OIDC Token endpoint.
+permissions:
+  id-token: write
```

Comment:
Why is write access needed?

Reply:
Setting the permissions to write is required in order to request an OpenID Connect JWT.

https://stackoverflow.com/questions/72183048/what-is-the-permission-scope-of-id-token-in-github-action

Removing this permission causes logging to fail to assume the token.

Reply:
We need a token in order to post the log to cloudwatch, I assume?

Reply:
Yes that's correct.

Reply:
This utilizes the same framework that AppSec signed off several months ago when we just started sending Cloudwatch logs from our canaries.